src/components/Analyze/AnalyzeList.tsx

```diff
@@ -1,12 +1,31 @@
+import { useEffect, useState, useContext } from 'react'
 import { AnalyzeContext } from '@/components/Analyze/AnalyzeContext'
 import { addComma } from '@/utils/money'
-import { useContext } from 'react'
+import { chartColors } from '@/Mock/Mock'
 
-const color = ['#5fb1ff', '#fffc58', '#8fff5f', '#df0000']
+interface widthType {
+	[key: number]: string
+}
 
 export const AnalyzeList = () => {
 	const { progressBarWidth, consumptionCombinedData } =
 		useContext(AnalyzeContext)
+
+	const [animatedWidth, setAnimatedWidth] = useState<widthType>({})
+
+	//width 기존에 0 걸어놓고, 필요한 width로 변경해서 transition 출력
+	useEffect(() => {
+		if (!consumptionCombinedData) return
+		const updatedWidths = consumptionCombinedData.reduce<widthType>(
+			(acc, data) => {
+				acc[data.id] = `${progressBarWidth ? progressBarWidth[data.id] : 0}%`
+				return acc
+			},
+			{},
+		)
```

Comment:
`consumptionCombinedData` 의 falsy 값을 타입가드 할거라면, `progressBarWidth` 도 같이 처리해주면 아래 코드가 깔끔해질 것 같습니다.

그리고 제가 코드를 잘 이해한 것이 맞다면, `widthType` 에서 [id, width] 형식을 갖고 위의 코드에서 그 값을 반환하는 것 같습니다.
그렇다면 `reduce` 가 필요할까? 라는 생각이 듭니다. 단순히 [id, width] 값만을 하나씩 매핑한다면 `map` 이 더 명시적이고 좋지 않을까라는 생각입니다. 처음 코드를 봤을 때 합을 누적시키는 것처럼 보여서 이해하기 조금 힘들었습니다.

```suggestion
const updatedWidths = Object.fromEntries(
      consumptionCombinedData.map(data => [
        data.id,
        `${progressBarWidth[data.id] ?? 0}%`
      ])
    )
```

Reply:
fromEntries를 처음 보네요, 타입 가드 의견 동의합니다.
훨씬 깔끔하네요